### DISCUSSION TITLE
Can an action return a body and redirect at the same time?

### DISCUSSION THREAD

**Discussion by wvhulle** [sveltejs/kit#6206] [Q&A]
The documentation does mention in https://kit.svelte.dev/docs/routing#page-actions that it is possible to add a redirect to the return of a POST action:
```
return {
    location: url
};
```
However, what if you would like to return a body and a redirect at the same time? When I have to return a body I use `new Response(String(body))` but adding the `location` as in `new Response(String(body), {headers: {location: url}})` doesn't give a functioning redirect. How can I do both at the same time?

**Comment by vekunz** [comment]
I am not sure what use case such a thing could be helpful for. If you return a redirect, every browser and most HTTP clients will automatically follow this redirect. You will likely never see the return body everywhere.
Besides that, it would be semantically wrong. A 302/303 HTTP return code indicates, that the result of your query is somewhere else and not here.

**Reply by wvhulle** [reply]
Okay, so it is one of both return types. Not at the same time.

**Reply by vekunz** [reply]
Yes, either you return a 2xx/4xx/5xx status code with a body, or you return a 3xx status code with a location header.
